src/Magnum/ImGuiIntegration/Context.hpp

```diff
@@ -40,93 +40,161 @@
 #include "Magnum/ImGuiIntegration/Integration.h"
 #include "Magnum/ImGuiIntegration/Context.h"
 
+#define HAS_NEW_IMGUI_IO (IMGUI_VERSION_NUM >= 18615)
```

Comment:
If you prefix the macro and hide it from Doxygen you can leave it defined, without having to `#undef` it at the end, and without having to define it again in the `*.cpp` file:

```suggestion
#if !defined(DOXYGEN_GENERATING_OUTPUT) && IMGUI_VERSION_NUM >= 18615
#define MAGNUM_IMGUIINTEGRATION_HAS_NEW_IMGUI_IO
#endif
```

I also changed it to be defined only if it's true, mainly for consistency with other defines out there.

Reply:
Since `Context.cpp` includes `Context.h` before `imgui.h` I still need to duplicate this (otherwise `IMGUI_VERSION_NUM` isn't defined yet when we need it). Should I just move the into a `configure.h`?

Reply:
You mean `configure.h.cmake`, right? Yes, that's a good place actually. And since it's ignored by Doxygen you don't even need the `#if !defined(DOXYGEN_GENERATING_OUTPUT)` there.

Reply:
There aren't any CMake variables we need so it could just be `.h` in the source folder. But I can go through CMake for consistency.

Reply:
Um, there already *is* `configure.h.cmake`, needed for distinguishing static/dynamic build ;)